options/posix/generic/netdb-stubs.cpp

```diff
@@ -22,20 +27,70 @@ void endservent(void) {
 	__builtin_unreachable();
 }
 
-void freeaddrinfo(struct addrinfo *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+void freeaddrinfo(struct addrinfo *ptr) {
+	auto buf = (struct mlibc::ai_buf*) ptr - offsetof(struct mlibc::ai_buf, ai);
+	free(ptr->ai_canonname);
+	free(buf);
```

Comment:
This logic is consistent with other C libs? Looks a bit strange.

Reply:
Yeah this is consistent with what musl does, and valgrind doesn't complain about this.